src/main/java/uk/gov/hmcts/cmc/claimstore/events/DocumentUploadHandler.java

```diff
@@ -1,43 +1,163 @@
+import static uk.gov.hmcts.cmc.domain.models.ClaimDocumentType.CLAIM_ISSUE_RECEIPT;
+import static uk.gov.hmcts.cmc.domain.models.ClaimDocumentType.DEFENDANT_PIN_LETTER;
 import static uk.gov.hmcts.cmc.domain.models.ClaimDocumentType.DEFENDANT_RESPONSE_RECEIPT;
+import static uk.gov.hmcts.cmc.domain.models.ClaimDocumentType.SEALED_CLAIM;
+import static uk.gov.hmcts.cmc.domain.models.ClaimDocumentType.SETTLEMENT_AGREEMENT;
 
 @Component
+@ConditionalOnProperty(prefix = "document_management", name = "url")
 public class DocumentUploadHandler {
 
-    private final ApplicationEventPublisher publisher;
+    private static final Logger logger = LoggerFactory.getLogger(DocumentUploadHandler.class);
+    private static final String CLAIM_MUST_NOT_BE_NULL = "Claim must not be null";
+
+    private final DocumentsService documentService;
     private final DefendantResponseReceiptService defendantResponseReceiptService;
+    private final CountyCourtJudgmentPdfService countyCourtJudgmentPdfService;
+    private final SettlementAgreementCopyService settlementAgreementCopyService;
+    private final SealedClaimPdfService sealedClaimPdfService;
+    private final ClaimIssueReceiptService claimIssueReceiptService;
+    private final DefendantPinLetterPdfService defendantPinLetterPdfService;
 
     @Autowired
-    public DocumentUploadHandler(ApplicationEventPublisher publisher,
-                                 DefendantResponseReceiptService defendantResponseReceiptService) {
-        this.publisher = publisher;
+    public DocumentUploadHandler(DefendantResponseReceiptService defendantResponseReceiptService,
+                                 CountyCourtJudgmentPdfService countyCourtJudgmentPdfService,
+                                 SettlementAgreementCopyService settlementAgreementCopyService,
+                                 SealedClaimPdfService sealedClaimPdfService,
+                                 ClaimIssueReceiptService claimIssueReceiptService,
+                                 DefendantPinLetterPdfService defendantPinLetterPdfService,
+                                 DocumentsService documentService) {
         this.defendantResponseReceiptService = defendantResponseReceiptService;
+        this.countyCourtJudgmentPdfService = countyCourtJudgmentPdfService;
+        this.settlementAgreementCopyService = settlementAgreementCopyService;
+        this.sealedClaimPdfService = sealedClaimPdfService;
+        this.claimIssueReceiptService = claimIssueReceiptService;
+        this.defendantPinLetterPdfService = defendantPinLetterPdfService;
+        this.documentService = documentService;
+    }
+
+    @EventListener
+    public void uploadDocument(CitizenClaimIssuedEvent event) {
+        Claim claim = event.getClaim();
+        requireNonNull(claim, CLAIM_MUST_NOT_BE_NULL);
+        PDF sealedClaim = new PDF(buildSealedClaimFileBaseName(event.getClaim().getReferenceNumber()),
```

Comment:
DocumentGenerator does react to CitizenClaimIssuedEvent and create all these documents and create a DocumentGeneratedEvent. Is It possible for us to add document upload to that class? While reviewing I can see similarities in functionality. Thus asking. May be wrong.

Reply:
Yeah , the DocumentGenerator actually issues events that send staff notifications and rpa etc. The name of the class is a bit misleading.